### PR TITLE
Comparing only the stateVersion in a pre-check of ledger extension to commit

### PR DESCRIPTION
**Pull request by jakrawcz-rdx**
This avoids the "Silently re-request useless LedgerSync from the same non-matching-ledger Node" (see https://rdxworks.slack.com/archives/C01M90Y2448/p1689936487103519?thread_ts=1689864950.471299&cid=C01M90Y2448)

### CODE REVIEW

**Comment by jakrawcz-rdx** [comment]
That comparator impl is probably not needed at all right now (used only in tests), but let's address this later, in develop.

File: core/src/main/java/com/radixdlt/ledger/StateComputerLedger.java [hunk 1]
```diff
@@ -326,7 +327,11 @@ private void commit(LedgerExtension ledgerExtension, VertexStoreState vertexStor
     synchronized (this.commitAndAdvanceLedgerLock) {
       final LedgerProof againstLedgerHeader = this.currentLedgerHeader;
 
-      if (this.headerComparator.compare(nextHeader, againstLedgerHeader) <= 0) {
+      if (nextHeader.getStateVersion() <= againstLedgerHeader.getStateVersion()) {
+        log.trace(
+            "ignoring the ledger extension {} which would not progress the current ledger {}",
```
Comment (dhedey) [inline core/src/main/java/com/radixdlt/ledger/StateComputerLedger.java:332]: Ignoring*

**Review by dhedey — APPROVED** [review]
Want to check that the tests pass ideally, just in case...

File: core/src/main/java/com/radixdlt/ledger/StateComputerLedger.java (hunk 1 above)
Comment (jakrawcz-rdx) [reply core/src/main/java/com/radixdlt/ledger/StateComputerLedger.java:332]: ok, let me wait with this push (let's not cancel the CI)

**Comment by jakrawcz-rdx** [comment]
Only the `MultiNodeRebootTest` failed.
I pushed the typo change.
Feel free to merge whenever.

**Pull request merged**